tests/core/pm-module/test_registry_integration.py

```diff
@@ -0,0 +1,200 @@
+    vy_registry.release(b"package1", b"1.0.1", b"p1.v1.0.1.com")
+    all_pkgs_1 = vy_registry.get_all_package_ids()
+    assert all_pkgs_1 == ((b'package1', 2),)
+    vy_registry.release(b"package2", b"1.0.0", b"p2.v1.0.0.com")
+    vy_registry.release(b"package3", b"1.0.0", b"p3.v1.0.0.com")
+    vy_registry.release(b"package3", b"1.0.1", b"p3.v1.0.1.com")
+    all_pkgs_2 = vy_registry.get_all_package_ids()
+    assert all_pkgs_2 == (
+        (b'package1', 2),
+        (b'package2', 1),
+        (b'package3', 2),
+    )
+    vy_registry.release(b"package4", b"1.0.0", b"p2.v1.0.0.com")
+    vy_registry.release(b"package5", b"1.0.0", b"p2.v1.0.0.com")
+    vy_registry.release(b"package6", b"1.0.0", b"p2.v1.0.0.com")
+    vy_registry.release(b"package7", b"1.0.0", b"p2.v1.0.0.com")
+    all_pkgs_3 = vy_registry.get_all_package_ids()
+    assert all_pkgs_3 == (
+        (b'package1', 2),
+        (b'package2', 1),
+        (b'package3', 2),
+        (b'package4', 1),
+        (b'package5', 1),
+        (b'package6', 1),
+        (b'package7', 1),
+    )
+
+
+@pytest.mark.skipif(ethpm_installed is False, reason="ethpm is not installed")
+def test_registry_get_all_package_versions(vy_registry):
+    vy_registry.release(b"package", b"1.0.0", b"p1.v1.0.0.com")
+    vy_registry.release(b"package", b"1.0.1", b"p1.v1.0.1.com")
+    vy_registry.release(b"package", b"1.0.2", b"p1.v1.0.2.com")
+    vy_registry.release(b"package", b"1.1.0", b"p1.v1.1.0.com")
+    vy_registry.release(b"package", b"2.0.0", b"p1.v2.0.0.com")
+    all_rls_1 = vy_registry.get_all_package_versions(b"package")
+    assert all_rls_1 == (
+        (b'1.0.0', b'p1.v1.0.0.com'),
+        (b'1.0.1', b'p1.v1.0.1.com'),
+        (b'1.0.2', b'p1.v1.0.2.com'),
+        (b'1.1.0', b'p1.v1.1.0.com'),
+        (b'2.0.0', b'p1.v2.0.0.com'),
+    )
+    vy_registry.release(b"package", b"3.0.0", b"p1.v3.0.0.com")
+    vy_registry.release(b"package", b"4.0.0", b"p1.v4.0.0.com")
+    vy_registry.release(b"package", b"5.0.0", b"p1.v5.0.0.com")
+    all_rls_2 = vy_registry.get_all_package_versions(b'package')
+    assert all_rls_2 == (
+        (b'1.0.0', b'p1.v1.0.0.com'),
+        (b'1.0.1', b'p1.v1.0.1.com'),
+        (b'1.0.2', b'p1.v1.0.2.com'),
+        (b'1.1.0', b'p1.v1.1.0.com'),
+        (b'2.0.0', b'p1.v2.0.0.com'),
+        (b'3.0.0', b'p1.v3.0.0.com'),
+        (b'4.0.0', b'p1.v4.0.0.com'),
+        (b'5.0.0', b'p1.v5.0.0.com'),
+    )
+
+
+@pytest.mark.skipif(ethpm_installed is False, reason="ethpm is not installed")
+def test_registry_transfer_owner(vy_registry, w3):
+    vy_registry.transfer_owner(w3.eth.accounts[1])
+    assert vy_registry.registry.functions.owner().call() == w3.eth.accounts[1]
+
+
+# web3.pm integration
+@pytest.mark.skipif(ethpm_installed is False, reason="ethpm is not installed")
+def test_pm_deploy_and_set_registry(w3):
+    w3.pm.deploy_and_set_registry()
+    assert isinstance(w3.pm.registry, Registry)
```

Comment:
should probably have a pre-check before the deployment to sanity check it wasn't somehow already present.